### PR TITLE
Fix mods navigation

### LINKED ISSUE #561: SPA Transitions to Home Page Broken
**Describe the bug**
The home page loads fine if you load it from the server, but won't load if you navigate to it within the client. Single-Page-Application (SPA)/client-side transitions to /mods work fine.

**To Reproduce**
1. Go to /mods.
2. Click on the `Home` NavLink or the logo to navigate to the home page.
3. The SPA transition never finishes and the home page never loads.

**Expected behavior**
The home page should load regardless of whether it is navigated to initially or through a SPA transition.

**Additional context**
Follow-up to #546.

### PR DESCRIPTION
**Pull request by ShouvikGhosh2048**
Closes #561.

### CODE REVIEW

**Comment by ShouvikGhosh2048** [comment]
@otobot1

I think the reason why the navigation wasn't working was because there was an infinite loop of `useEffect`'s and setting states. Using `useMemo` fixes the navigation. However, the memo function still runs when we interact with the table (you can check the browser console - it prints 1,2,3,4,5.). I think the memoization isn't working as we're using arrays as dependencies.

**Comment by otobot1** [comment]
@ShouvikGhosh2048 Good catch! We'll need to fix that as well before launch. I'm hoping I'll have a bit of time this weekend to at least catch up on reviewing PRs - I managed to get through all of the dependendabot ones finally.

**Comment by ShouvikGhosh2048** [comment]
@otobot1

https://github.com/celestemods-com/CelesteMods/blob/7cbc34a02a3093c87e7f055303977f3d001533fb/src/pages/mods/index.tsx#L277-L284

`ratingQueries` gets a new value on each render and thus the memos are recalculated. Should it be running on every render?

**Comment by otobot1** [comment]
> @otobot1
> 
> https://github.com/celestemods-com/CelesteMods/blob/7cbc34a02a3093c87e7f055303977f3d001533fb/src/pages/mods/index.tsx#L277-L284
> 
> `ratingQueries` gets a new value on each render. It is used as a dependency and thus the memos are recalculated. Should it be running on every render?

Nooope. That only needs to re-run periodically. Maybe every 30 seconds? It also probably shouldn't be directly used as a depencency I guess. We only want everything to re-render when the actual data returned by query changes, not every time we poll the server to check if anything has changed. I didn't explore the [React Query docs](https://tanstack.com/query/v4/docs/react/overview) very deeply when I was writing that, so maybe it's just configured wrong. I suppose it could also be something misconfigured in the [tRPC client](https://trpc.io/docs/client/react), but that's a pretty thin wrapper around React Query so probably not.

**Comment by ShouvikGhosh2048** [comment]
Since we only care about the data, we could do something like this:

```JS
    const newRatingQueriesData = ratingQueries.map(rating => rating.data);
    const [oldRatingQueriesData, setOldRatingQueriesData] = useState<typeof newRatingQueriesData>([]);

    let ratingQueriesData: typeof newRatingQueriesData;
    if (JSON.stringify(oldRatingQueriesData) !== JSON.stringify(newRatingQueriesData)) {
        setOldRatingQueriesData(newRatingQueriesData);
        ratingQueriesData = newRatingQueriesData;
    } else {
        ratingQueriesData = oldRatingQueriesData;
    }
```
and then use `ratingQueriesData` as a dependency. This seems to work.

**Comment by ShouvikGhosh2048** [comment]
I guess another option might be to create a `<Query>` component which contains all the queries, uses `<Mods>` as a child component, and passes the queries to `<Mods>` using props. Then state change in `<Mods>` won't recreate the `ratingQueries`. Which option do you prefer?

**Comment by otobot1** [comment]
I don't have a strong preference, but I suspect the Query component may be more performant, so let's go with that.

File: src/utils/api.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export const api = createTRPCNext<AppRouter>({
         }),
         httpBatchLink({
           url: `${getBaseUrl()}/api/trpc`,
+          maxURLLength: 2083,
```
Comment (ShouvikGhosh2048) [inline src/utils/api.ts:44]: https://trpc.io/docs/client/links/httpBatchLink#setting-a-maximum-url-length

This was an error occurring in the website.

Comment (otobot1) [reply src/utils/api.ts:44]: 👍

**Comment by otobot1** [comment]
Looks like the problem is fixed for the mods table! I prefer to give each component its own file, so I extracted it into [/components/mods/modsTable.tsx](https://github.com/ShouvikGhosh2048/CelesteMods/blob/fix-mods-navigation/src/components/mods/modsTable.tsx). 

When I opened up an expanded mod, the flood of errors restarted. It looks like it's the same problem, so please create a query parent for the maps table as well.

Follow-up issue: I noticed that every time I click out of the page (into another window, or even just out of the page and into the browser console or settings) and then back in, the queries re-run and emit new network requests. This needs to be limited to a more reasonable rate.

**Comment by ShouvikGhosh2048** [comment]
I've create the query parent.

**Comment by otobot1** [comment]
Thanks!

**Pull request merged**